From 01379d07e6084453461eb48ea83fe3982bf08bd1 Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Thu, 23 Feb 2017 16:31:37 +0800
Subject: [PATCH 1538/5242] MLK-14252-2 video: mipi_dsi_northwest: reset dsi
 domains when dsi disabled.

commit  5ccbcf78264f65dbfdcc726c604ab143545e715c from
https://source.codeaurora.org/external/imx/linux-imx.git

Put the three dsi domains into reset state when dsi is disabled
to avoid dsi being an unstable state before next time enabled.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/video/fbdev/mxc/mipi_dsi_northwest.c |    6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/video/fbdev/mxc/mipi_dsi_northwest.c b/drivers/video/fbdev/mxc/mipi_dsi_northwest.c
index 6aa31bb..fb30d0f 100644
--- a/drivers/video/fbdev/mxc/mipi_dsi_northwest.c
+++ b/drivers/video/fbdev/mxc/mipi_dsi_northwest.c
@@ -512,6 +512,8 @@ static int mipi_dsi_enable(struct mxc_dispdrv_handle *disp,
 			return -EINVAL;
 		}
 
+		reset_dsi_domains(mipi_dsi, 0);
+
 		ret = mipi_display_exit_sleep(mipi_dsi->disp_mipi);
 		if (ret) {
 			dev_err(&mipi_dsi->pdev->dev, "exit sleep failed\n");
@@ -703,6 +705,10 @@ static void mipi_dsi_disable(struct mxc_dispdrv_handle *disp,
 	}
 
 	clk_disable_unprepare(mipi_dsi->esc_clk);
+
+	reset_dsi_domains(mipi_dsi, 1);
+	regmap_update_bits(mipi_dsi->regmap, SIM_SOPT1CFG,
+			   DSI_PLL_EN, 0x0);
 }
 
 static int mipi_dsi_setup(struct mxc_dispdrv_handle *disp,
-- 
1.7.9.5

